task/oci-copy/0.1/oci-copy.yaml

```diff
@@ -0,0 +1,157 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: image-build, appstudio, hacbs
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: oci-artifact
+  name: oci-copy
+spec:
+  description: |-
+    TODO
+  params:
+    - description: Reference of the image we will push
+      name: IMAGE
+      type: string
+    - default: ./oci-copy.yaml
+      description: Path to the oci copy file.
+      name: OCI_COPY_FILE
+      type: string
+  results:
+    - description: Digest of the artifact just pushed
+      name: IMAGE_DIGEST
+    - description: Repository where the artifact was pushed
+      name: IMAGE_URL
+  stepTemplate:
+    env:
+      - name: OCI_COPY_FILE
+        value: $(params.OCI_COPY_FILE)
+      - name: IMAGE
+        value: $(params.IMAGE)
+    volumeMounts:
+      - mountPath: "/var/workdir"
+        name: workdir
+  steps:
+    - name: prepare
+      image: quay.io/konflux-ci/yq:latest@sha256:974dea6375ee9df561ffd3baf994db2b61777a71f3bcf0050c5dca91ac9b3430
+      script: |
+        set -eu
+        set -o pipefail
+
+        SOURCE_CODE_DIR=source
+        oci_copy_file_path="$(pwd)/$SOURCE_CODE_DIR/$OCI_COPY_FILE"
+
+        mkdir -p /var/workdir/vars/
+
+        for entry in $(cat $oci_copy_file_path | yq '.artifacts[] | @json | @base64'); do
+          entry=$(echo $entry | base64 -d)
+          source=$(echo $entry | yq .source)
+          filename=$(echo $entry | yq .filename)
+          artifact_type=$(echo $entry | yq .type)
+          artifact_digest=$(echo $entry | yq .sha256sum)
+
+          echo "declare OCI_SOURCE=${source}" > /var/workdir/vars/$filename
+          echo "declare OCI_FILENAME=${filename}" >> /var/workdir/vars/$filename
+          echo "declare OCI_ARTIFACT_TYPE=${artifact_type}" >> /var/workdir/vars/$filename
+          echo "declare OCI_ARTIFACT_DIGEST=${artifact_digest}" >> /var/workdir/vars/$filename
+
+          echo "Wrote /var/workdir/vars/$filename with contents:"
+          cat /var/workdir/vars/$filename
+        done
+      workingDir: $(workspaces.source.path)
+
+    - name: oci-copy
+      # I need something with adequate rights to run:
+      # $ skopeo copy docker://quay.io/containers/buildah:v1.35.4 docker://quay.io/redhat-appstudio/buildah:v1.35.4
```

Comment:
Copied. I suppose this is a quick workaround and we need to start building our own buildah? (I wish we could use the redhat buildah, but apparently the release cadence doesn't meet our needs)

Reply:
Yeah! Hey, I was supposed to join the containres cabal meeting last week to talk about onboarding upstream buildah builds to konflux (the container tools team would actually own the builds), but we had to cancel.

Assuming that goes forward - I wonder if we'd be happy just consuming their builds. Or, letting them do a double-release, one to their quay namespace but also to ours? :thinking: 

In anycase, this skopeo copy is just short-term, yes.

Reply:
> I wonder if we'd be happy just consuming their builds.

I'd be happy if we could be happy with that :grin: 

I'm thinking we would need a CI check that verifies the attestations of images outside the `quay.io/konflux-ci` org and we'd have a list of Chains signing keys or workflow identities etc. to trust

Reply:
Yeah. Let's aim in that direction. :)

Reply:
Update `image` ref in 1749ad3.